### DISCUSSION TITLE
The custom dictionary feature suddenly stop working

### DISCUSSION THREAD

**Discussion by ooker777** [tadashi-aikawa/obsidian-various-complements-plugin#262] [Q&A] [answered]
I open the sandbox vault, delete every note and create `dictionary.md` with this content:
```markdown
aaa		nc
```
I:
- set both `dictionary.md` and `./dictionary.md` as the path, 
- turn off all other features and fuzzy match, 
- turn on auto-complement 
- set min number of triggering characters to 0, 
- use prefix match strategy 
![](https://i.imgur.com/dpMraIC.png)

I reload the custom dictionary, then open the console. I create another note and type in `nc`. There is no suggestion as expected. Why is that?

**Comment by tadashi-aikawa** [comment]
@ooker777 
Could you tell me your settings copied by the [Copy plugin settings](https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/3.%20Commands/%F0%9F%8E%AE%20Copy%20plugin%20settings/) command?

**Reply by ooker777** [reply]
```json
{
    "version": "9.0.1",
    "mobile": false,
    "settings": {
        "strategy": "default",
        "cedictPath": "./cedict_ts.u8",
        "matchStrategy": "prefix",
        "fuzzyMatch": false,
        "minFuzzyMatchScore": 0.5,
        "matchingWithoutEmoji": true,
        "treatAccentDiacriticsAsAlphabeticCharacters": false,
        "maxNumberOfSuggestions": 5,
        "maxNumberOfWordsAsPhrase": 3,
        "minNumberOfCharactersTriggered": 0,
        "minNumberOfWordsTriggeredPhrase": 1,
        "complementAutomatically": true,
        "delayMilliSeconds": 0,
        "disableSuggestionsDuringImeOn": false,
        "insertAfterCompletion": true,
        "firstCharactersDisableSuggestions": ":/^",
        "patternsToSuppressTrigger": [
            "^~~~.*",
            "^```.*"
        ],
        "noAutoFocusUntilCycle": false,
        "showMatchStrategy": true,
        "showComplementAutomatically": true,
        "showIndexingStatus": true,
        "descriptionOnSuggestion": "Short",
        "hotkeys": {
            "select": [
                {
                    "modifiers": [],
                    "key": "Enter"
                }
            ],
            "up": [
                {
                    "modifiers": [],
                    "key": "ArrowUp"
                }
            ],
            "down": [
                {
                    "modifiers": [],
                    "key": "ArrowDown"
                }
            ],
            "select 1st": [],
            "select 2nd": [],
            "select 3rd": [],
            "select 4th": [],
            "select 5th": [],
            "select 6th": [],
            "select 7th": [],
            "select 8th": [],
            "select 9th": [],
            "open": [],
            "completion": [],
            "insert as text": []
        },
        "propagateEsc": false,
        "enableCurrentFileComplement": false,
        "currentFileMinNumberOfCharacters": 0,
        "onlyComplementEnglishOnCurrentFileComplement": false,
        "enableCurrentVaultComplement": false,
        "currentVaultMinNumberOfCharacters": 0,
        "includeCurrentVaultPathPrefixPatterns": "",
        "excludeCurrentVaultPathPrefixPatterns": "",
        "includeCurrentVaultOnlyFilesUnderCurrentDirectory": false,
        "enableCustomDictionaryComplement": true,
        "customDictionaryPaths": "dic.md",
        "columnDelimiter": "Tab",
        "customDictionaryWordRegexPattern": "",
        "delimiterToHideSuggestion": "",
        "delimiterToDivideSuggestionsForDisplayFromInsertion": "",
        "caretLocationSymbolAfterComplement": "",
        "displayedTextSuffix": " => ...",
        "enableInternalLinkComplement": false,
        "suggestInternalLinkWithAlias": false,
        "excludeInternalLinkPathPrefixPatterns": "",
        "updateInternalLinksOnSave": true,
        "insertAliasTransformedFromDisplayedInternalLink": {
            "enabled": false,
            "beforeRegExp": "",
            "after": ""
        },
        "frontMatterKeyForExclusionInternalLink": "",
        "enableFrontMatterComplement": false,
        "frontMatterComplementMatchStrategy": "inherit",
        "insertCommaAfterFrontMatterCompletion": false,
        "intelligentSuggestionPrioritization": {
            "enabled": false,
            "historyFilePath": "",
            "maxDaysToKeepHistory": 30,
            "maxNumberOfHistoryToKeep": 0
        },
        "disableOnMobile": false,
        "showLogAboutPerformanceInConsole": false,
        "selectionHistoryTree": {}
    }
}
```

**Comment by tadashi-aikawa** [comment]
@ooker777 
In your initial post, you wrote the following. 

> set both `dictionary.md` and `./dictionary.md` as the path,

However, the current setting is as follows.

> "customDictionaryPaths": "dic.md",

Therefore, you need to correct the setting.

![image](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/assets/9500018/beb53b86-7918-4d85-ad3a-1027953f159d)

**Reply by ooker777** [reply]
no, I used the correct path. I used the sandbox vault so after closing it I had to recreate the problem again 😅. I just didn't use the same filename. Too lazy to write the full word 😅

**Reply by tadashi-aikawa** [reply, marked as answer]
@ooker777 
OK! I could reproduce it and found out the reason. 

When I enabled the "Show log about performance in a console" option, a log was output explaining the reason.

![image](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/assets/9500018/5a2c9726-65c3-4e90-9b2e-d1ae12e7c35c)

```
[onTrigger] Don't show suggestions because currentToken is less than minNumberTriggered option
```

It is a specification of the [⚙️ Min number of characters for trigger](https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/4.%20Options/4.1.%20Main/%E2%9A%99%EF%B8%8F%20Min%20number%20of%20characters%20for%20trigger/) option.

![image](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/assets/9500018/ebe51558-cc6d-440c-9b71-9e800744258c)

**Reply by ooker777** [reply]
thanks. I work it out. This is the second time I got to this problem. I understand the rationale behind this, but apparently the documentation isn't clear enough for me. I make a PR for this, can you check? https://github.com/tadashi-aikawa/docs-obsidian-various-complements-plugin/pull/2

**Reply by tadashi-aikawa** [reply]
@ooker777 
Thank you for the pull request! I will review it.
